### ISSUE TITLE
Fedora 31 in dom0

### ISSUE THREAD

**Issue by marmarek** [QubesOS/qubes-issues#5529]

**Comment by JarrahG** [comment]
Please let me know when this is ready for testing and what will be required for testing. More than happy to test it out on the 4.1 system I have up.

**Comment by brendanhoar** [comment]
And I was just about to install 4.1 from the October iso ...

**Comment by marmarek** [comment]
Oh great, salt package in F31 is broken: https://bugzilla.redhat.com/show_bug.cgi?id=1748170

**Comment by marmarek** [comment]
Even better: F31 has pre-release of salt package (salt-2019.2.1rc0.3), which is affected by https://github.com/saltstack/salt/issues/54765 (fixed in 2019.2.2)

**Comment by marmarek** [comment]
Reported upstream: https://bugzilla.redhat.com/show_bug.cgi?id=1785489

**Comment by marmarek** [comment]
Unfortunately the above makes initial setup completely broken. I guess it means no installation image before Xmas  :/

**Comment by fepitre** [comment]
Maybe we could do the update on  upstream?

**Comment by marmarek** [comment]
Yes, submitting a pull request would probably help them. In worst case, we can ship the update through our repos...

**Comment by fepitre** [comment]
@marmarek: would you like me to switch all the travis dom0 to fc31?

**Comment by marmarek** [comment]
> @marmarek: would you like me to switch all the travis dom0 to fc31?

Generally, only after all the required changes are in. But you can add relevant changes to fc31 PRs. Please hold off with creating PRs for other repos until those are merged and default builder.conf is updated, to not clutter pull request list.

**Comment by marmarek** [comment]
Latest build is here: https://openqa.qubes-os.org/tests/5071/asset/iso/Qubes-4.1-20191222-x86_64.iso
It's build using this config: https://gist.github.com/marmarek/05f182e51d7cd1872aaea6b34c936920
Major changes since last test build are: Fedora 31 in dom0 and Xen 4.13.

It is not tested at all, besides the installer itself.
One of the changes in Xen 4.13 is dropped support for PCI passthrough without IOMMU, even for PV domains. This means it is no longer possible to run Qubes on hardware without IOMMU, even with switching virt_mode to pv.
This, sadly, also affect our OpenQA instance, which don't have IOMMU inside KVM...

**Comment by JarrahG** [comment]
Installed and quickly tested. I have noted the following:

Installer (UEFI):

Install went fine until bootloader. At this point anaconda noted an
error in installing the bootloader. The correct files were in
/boot/efi/EFI/qubes except for grub.cfg. Manually generating this file
did not fix the problem, however, it showed that anaconda had set the
boot file to /boot/efi/EFI/fedora/grub.efi. Moving all Qubes install
files to the fedora folder got a working boot. (This includes the
manually generated grub.cfg, not sure if fixing the above folder issue
will fix the grub config one).

First boot:

* First install failed and rebooted, then failed again with the attached
error message. On clicking through this message, the system was
installed and working fine with a fedora-30 template, so the
configuration worked but still errored.

* Sys-usb was not an option to configure due to a USB keyboard being
present. However, this machine uses a PS/2 keyboard and a USB mouse.

* Both #5519 and #5517 are fixed in this version.

**Comment by marmarek** [comment]
Thanks for testing @JarrahG !
New build, with (hopefully) UEFI issue fixed: https://openqa.qubes-os.org/tests/5110/asset/iso/Qubes-4.1-20191223-x86_64.iso
The other issue I'm not sure. The message you've got is result of running through firstboot twice. But the reason for the first failure is still a mystery.

**Comment by tmp-jarrahG** [comment]
Tested the above version. Most of the above is fixed here. 
* Grub installs correctly to the /boot/efi/EFI/qubes folder. 
* Generation of grub.cfg works automatically. 
* First boot works first time. Though looks odd. See below. 

However, the following issues still occur or are new:
* EFI boot is still configured for EFI/fedora/shimx64.efi rather than EFI/qubes/grubx64.efi. This is likely in the call to `efibootmgr` and is easily manually fixed. 
* Sys-usb is still not available in first boot. 
* The first boot install doesn't fill the whole screen. It takes up only a small section in the top left. However, it is still usable. 
* Creating sys-usb using `qubesctl` results in multiple python warnings but otherwise works:
```
[WARNING ] /usr/lib/python3.7/site-packages/salt/states/file.py:294: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  from collections import Iterable, Mapping, defaultdict

[WARNING ] /usr/lib/python3.7/site-packages/salt/utils/decorators/signature.py:31: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
  *salt.utils.args.get_function_argspec(original_function)

[WARNING ] /usr/lib/python3.7/site-packages/salt/utils/decorators/signature.py:31: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
  *salt.utils.args.get_function_argspec(original_function)

[WARNING ] /usr/lib/python3.7/site-packages/salt/utils/decorators/signature.py:31: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
  *salt.utils.args.get_function_argspec(original_function)

[WARNING ] /usr/lib/python3.7/site-packages/salt/utils/oset.py:47: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  class OrderedSet(collections.MutableSet):
```


Along with both of the above issues, which are still solved in this version, #5422 is also fixed. 

- JarrahG, sent from this account as I cannot currently get to my normal one.

**Comment by tmp-jarrahG** [comment]
Tested todays version from openqa:
* EFI boot now works first go. 
* sys-usb is available in firstboot. I'll retest this on my other machine in a few days, but it works on the laptop. 
* Qubesctl python deprecations still occur, though this time I found them in /var/log/salt/minion and /var/log/anaconda/{syslog,journal.log}. These aren't really a problem until Qubes 4.2, assuming it is using something above Fedora 31. 
* no-real-mode and edd show up as unknown parameters to Xen in `xl dmesg`.

In 5 minutes of using it after install, it all seems quite usable. I'll do some more detailed testing in the next few days.

**Comment by xaki23** [comment]
"no-real-mode" is/was mostly relevant for people booting through kexec (linuxboot, heads,...)
if this is no longer required (because xen figures out what mode the cpu is in on its own), good.
if it is still required (but gone), bad.

**Comment by firstabor** [comment]
Is Linux Kernel 5.3+ working on that ISO?

**Comment by marmarek** [comment]
> if it is still required (but gone), bad.

I think the message is false positive. Handling of those two options looks different from other cases.

>     * kernel-latest seems to be missing from the repo.

Repositories are already set for fc31, but the packages are still in fc29. You can try changing temporarily the URL. But be careful to not install other (really fc29 targeted) packages.

**Comment by firstabor** [comment]
Tried to install 
https://openqa.qubes-os.org/tests/5110/asset/iso/Qubes-4.1-20191223-x86_64.iso

But Qubes is not booting after restart boots back into installer again unpluggin the usb and booting from internal storage gives me the error

/boot/EFI/fedora/grubx64.efi missing

**Comment by marmarek** [comment]
This issue should be fixed in a newer build: https://openqa.qubes-os.org/tests/5153/asset/iso/Qubes-4.1-20191225-x86_64.iso

**Comment by marmarek** [comment]
I'm preparing new build with 5.4 kernel.

**Comment by firstabor** [comment]
> I'm preparing new build with 5.4 kernel.
Trying to install the fix now


Nice when do you think you will have it ready?

> This issue should be fixed in a newer build: https://openqa.qubes-os.org/tests/5153/asset/iso/Qubes-4.1-20191225-x86_64.iso

Trying to install the fix now.

When will you be ready with the 5.4 build?

**Comment by marmarek** [comment]
> * First install failed and rebooted, then failed again with the attached error message.

I've managed to reproduce this (or similar issue). Looks like a dom0 kernel crash on a DISCARD operation. See below.
The issue seems to be fixed in 5.4 kernel.

![IMG_20191227_143845-](https://user-images.githubusercontent.com/726704/71526351-bbf92080-28d6-11ea-8821-e6eb7133e811.jpg)

> When will you be ready with the 5.4 build?

The build is running already, should finish in about an hour.

**Comment by brendanhoar** [comment]
Looks like build might be done about 20 minutes ago: https://openqa.qubes-os.org/tests/5242/asset/iso/Qubes-4.1-20191227-x86_64.iso

Tests have just started...

**Comment by brendanhoar** [comment]
Installed Qubes-4.1-20191227 cleanly on a Thinkpad P52. BIOS changes were: disabled Ethernet and Wifi just in case for installation, and changed Display from Hybrid/Optimus to Discrete (Nvidia).

Display is 4K, so when I got to the firstboot screen, I had to blindly navigate (using tab key) a really small window in the upper left hand corner...

![image](https://user-images.githubusercontent.com/8979139/71537844-d1c70f80-28ef-11ea-9911-7c9b16b19a0d.png)

...which worked, but is non-optimal.

After install finished, I enabled WiFi and attached it to sys-net.

In any case, now Qubes 4.1 (test) with kernel 5.4.x is installed and mostly working on a Thinkpad P52.

B

**Comment by 0spinboson** [comment]
> PS - used xfwm Display control panel to change to 1920x1080 because...old eyes. Display turned black. ctrl-alt-f2 followed by ctrl-alt-f1 brought me back to a working X windows. Seems like I might need to do this regularly when the screen turns black. xrandr in domU VMs so far seem to report 1920x1080 correctly.
> 
> PPS - video playback is sooooper-laggy. Way worse than the optimized setup I have on a Thinkpad W520 (2011 era). So, tweaking looks inevitable. Need to figure out where vertical-sync, etc. are hiding...

Could these be nvidia driver issues? I dunno (though doubt) the quadro p2000 is well-supported by the foss driver.

**Comment by tmp-jarrahG** [comment]
This build seems to break the fix for https://github.com/QubesOS/qubes-issues/issues/5422 found in the previous 4.1 builds. Not sure exactly what is failing, but I'll be trying to get a hypervisor console shortly. Currently, Xen boots, but linux never displays any output. 

In previous builds I have also had the above mini-first boot screen. But it was workable with a mouse. Also using a 4K screen.

**Comment by GWeck** [comment]
See also comments on using the 23 Dec ISO in #5003

**Comment by brendanhoar** [comment]
> Could these be nvidia driver issues? I dunno (though doubt) the quadro p2000 is well-supported by the foss driver.

Probably. Need to figure out how to get rpmfusion set up, doesn't seem to be in the baseline config (even commented out).

**Comment by JarrahG** [comment]
Tested on my other machine. This one also has a 4K screen, but didn't display the mini first boot problem shown above. Screen resolution may be unrelated to getting this to trigger. 

Also worked out what the sys-usb problem was this time. My mouse has an emulated keyboard which triggers the issue. That one is a separate issue.

**Comment by zaoqi** [comment]
@marmarek 
> One of the changes in Xen 4.13 is dropped support for PCI passthrough without IOMMU, even for PV domains. This means it is no longer possible to run Qubes on hardware without IOMMU, even with switching virt_mode to pv.
> This, sadly, also affect our OpenQA instance, which don't have IOMMU inside KVM...

The OpenQA instance doesn't seem to be affected

https://openqa.qubes-os.org/tests/5259#step/mount_and_boot_options/15

**Comment by marmarek** [comment]
If you look closely, there is no `hvm_directio` nor `pv_directio` in `virt_caps`. It means it isn't possible to start any VM with PCI devices assigned (in this Xen version, not even PV). For the time being, I've made a hack to start sys-usb/sys-net without any PCI devices, just to have the initial setup not crashing. I'll need to figure out some better solution for other tests.

**Comment by 92VV3M42d3v8** [comment]
Installation bugs down to soft lock up.

**Comment by 92VV3M42d3v8** [comment]
Does bug in sys-USB haven't removed from these builds too?

**Comment by JarrahG** [comment]
Could we get a screenshot/more detail on the CPU lockup? The Nvidia graphics may have something to do with it, most of the newer ones aren't supported. 

Is this the sys-usb bug I was talking about above where the first boot screen doesn't allow the creation on one? If so, that was a hardware issue on my end (mouse reports as a keyboard), not a Qubes issue.

**Comment by marmarek** [comment]
Have anybody tried system suspend with those test images? The one system I've tried (with UEFI boot) resets on resume.

**Comment by brendanhoar** [comment]
@marmarek - On the P52 resuming from suspend also led to a reboot. Non-UEFI.

**Comment by JarrahG** [comment]
Tested on two systems: 
Desktop AMD system doesn't suspend on either 4.0 or 4.1. Screen goes blank and VMs stop (seen by usb controllers stopping) but the fans still run and the system doesn't indicate a suspend. 

Laptop running the 2019-12-25 build (27 doesn't boot) has the issue described by @marmarek above. Suspends fine, reset on resume. 

The guys here also tested suspend and had some issues on both 4.0 and 4.1. https://groups.google.com/d/msg/qubes-users/67LmZ8LsR9A/RkRZAYLsBwAJ

**Comment by zaoqi** [comment]
@marmarek 
> For the time being, I've made a hack to start sys-usb/sys-net without any PCI devices, just to have the initial setup not crashing. I'll need to figure out some better solution for other tests.

Allowing some users to use legacy xen might be a solution

**Comment by marmarek** [comment]
> Allowing some users to use legacy xen might be a solution

That's not an option. Systems without IOMMU are not supported by Qubes 4.0+, as announced few years ago already. The fact it was possible to break it in 4.0 (severely reducing isolation), doesn't mean we should ease it. Note new Xen, besides this requirement also have many improvement we will take advantage of.

**Comment by 0spinboson** [comment]
perhaps offtopic, but is the openqa server currently down?

**Comment by marmarek** [comment]
> perhaps offtopic, but is the openqa server currently down?

Looks like it.

**Comment by xaki23** [comment]
related: freenode, #qubes, 2019-12-31

17:24 < dmoerner> Also, it looks like openqa is down?
17:25 < dnull> Shit
17:25 < dnull> 1 sec
17:25 < dnull> I'll check it
17:25 < dnull> We had issues at the DC last night

**Comment by JarrahG** [comment]
Having issues updating. Calling the standard `qubes-dom0-update` and enabling the current testing repo, dnf reports that it cannot find `file:///var/lib/qubes/updates/[...]/repomd.xml`. Not sure if this is just because there are no updates to be had or there is a bug in the script.

**Comment by 0spinboson** [comment]
I just installed the 27 dec build locally. Installation proceeded without issues, including initial configuration. Resuming after suspend didn't result in reboot, but did result in an unresponsive system.
Only tweak I'd suggest: reducing default max_mem use for sys-firewall from 4gb to 500-600MB. :)

**Comment by githubhun1** [comment]
I have installed the 23 Dec iso (legacy boot, AMD fam15h laptop with coreboot) and here are my observations:
- I had to fix the xen-discard issue (reboot on qvm-shutdown guest) by installing kernel-latest, and then running the initial qube setup manually
- from then on everything so far works as expected
- suspend/resume: on some AMD systems there is already an issue with dom0 suspend/resume from xen-4.8.3-3 onward that the CPU capability bits for the boot CPU(0) change (cap[1], bits 27 & 31) after resume causing a xen panic (in xen/arch/x86/acpi/power.c). Some recent details are here: https://groups.google.com/forum/#!topic/qubes-users/67LmZ8LsR9A. Workaround is to disable the features check.  Now, with xen-4.13 (patched power.c) this is not working any more, as there are additional checks for non-boot cpus:

[2020-01-03 11:00:04] (XEN) Entering ACPI S3 state.
[2020-01-03 11:00:04] (XEN) AMD-Vi: Applying erratum 746 workaround for IOMMU at 0000:00:00.2
[2020-01-03 11:00:04] (XEN) Finishing wakeup from ACPI S3 state.
[2020-01-03 11:00:04] (XEN) CPU0: cap[ 1] is 3e98320b (expected b698320b)
[2020-01-03 11:00:04] (XEN) Missing previously available feature(s).
[2020-01-03 11:00:04] (XEN) Enabling non-boot CPUs  ...
[2020-01-03 11:00:04] (XEN) CPU1: cap[ 1] is 3e98320b (expected b698320b)
[2020-01-03 11:00:04] (XEN) CPU1: Failed to validate features - not coming back online
[2020-01-03 11:00:04] (XEN) Error bringing CPU1 up: -6
[2020-01-03 11:00:04] (XEN) CPU2: cap[ 1] is 3e98320b (expected b698320b)
[2020-01-03 11:00:04] (XEN) CPU2: Failed to validate features - not coming back online
[2020-01-03 11:00:04] (XEN) Error bringing CPU2 up: -6
[2020-01-03 11:00:04] (XEN) CPU3: cap[ 1] is 3e98320b (expected b698320b)
[2020-01-03 11:00:04] (XEN) CPU3: Failed to validate features - not coming back online
[2020-01-03 11:00:04] (XEN) Error bringing CPU3 up: -6
[2020-01-03 11:00:04] (XEN) Removing cpu 1 from runqueue 0
[2020-01-03 11:00:04] (XEN) Removing cpu 2 from runqueue 0
[2020-01-03 11:00:04] (XEN) Removing cpu 3 from runqueue 0

This results of having only the boot cpu online after resume.  Without the patch, the "usual" xen panic occurs on resume (screen off, fans are max, system unresponsive).

**Comment by githubhun1** [comment]
...continued...
I modified the recheck_cpu_features function (4.13), so the cpu cap high bits are masked out. Now all other code calling this check also runs through (so I have all 4 cpus back after resume).
Result: HVMs (sys-net, sys-usb) freeze after resume (--> qvm-kill), PVHs seem to resume OK.
For HVMs the stubdomain resumes, at least I still can login to those after resume and I don't see any issues in the stubdom logs.
After killing the HVMs, I can start sys-net and sys-usb again and those seem to work OK then.
Do I miss something - like installing a "compatible" xen-4.13 modded (not upstream) binary into the HVM templates? 
By the way I use already Fedora-31-xfce templates for these tests.

**Comment by brendanhoar** [comment]
@marmarek
> One of the changes in Xen 4.13 is dropped support for PCI passthrough without IOMMU, even for PV domains. This means it is no longer possible to run Qubes on hardware without IOMMU, even with switching virt_mode to pv.
> This, sadly, also affect our OpenQA instance, which don't have IOMMU inside KVM...

Is the issue that openqa does not support IOMMU via KVM or just that the current hosting hardware for OpenQA does not provide IOMMU inside KVM? 

B

**Comment by brendanhoar** [comment]
> Have anybody tried system suspend with those test images? The one system I've tried (with UEFI boot) resets on resume.

@marmarek 

Since it appears the old made-for-purpose USB 2.0 EHCI Debug port dongles are impossible to find these days, I've been looking around for alternatives and stumbled upon use of the raspberry pi zero w/ USB Gadget drivers to log chromebook coreboot debug data. Pretty sure (but not 100%) the same could be done for Xen debug data:

https://johnlewis.ie/pi-zero-w-flashrom-and-usb-gadget-debug/
https://johnlewis.ie/wp-content/uploads/2017/04/ehcidebug.gif

B

Speaking of debugging Xen resume failures

**Comment by 0spinboson** [comment]
I'm seeing more than a few issues with windows not being drawn properly, using the templates provided by the 12-27 installation medium, while others take a few seconds to fill/show the contents. Very consistently affecting notification windows (think browser notification popups).

**Comment by xaki23** [comment]
> Since it appears the old made-for-purpose USB 2.0 EHCI Debug port dongles are impossible to find these days, I've been looking around for alternatives and stumbled upon use of the raspberry pi zero w/ USB Gadget drivers to log chromebook coreboot debug data. Pretty sure (but not 100%) the same could be done for Xen debug data:

unsure if this might work.
i asked on #coreboot because i wasnt sure if this raspi setup might be just a software-suzyqable (it is not) and got warned that "ehci debug is limited to older devices with an actual ehci controller/ports".
and now i am still not sure if xen supports _any_ of that, even if you run it on the right retro hardware. 

what hardware/platform/device would you try this on?

**Comment by brendanhoar** [comment]
W520 and X230. I have some spares of each to test with, while primary system is still running Qubes R4.0.

I have a P52, but it doesn't appear to support EHCI debugging ... and I'm not sure if XHCI cables can be used with Xen

**Comment by marmarek** [comment]
I have one laptop with physical serial port (through a docking station) and extracted the logs: https://xen.markmail.org/thread/w5e47lvli7i3wjwu

> Is the issue that openqa does not support IOMMU via KVM or just that the current hosting hardware for OpenQA does not provide IOMMU inside KVM?

The issue is incomplete AMD IOMMU emulation support in KVM (no support for X2APIC if I understand correctly). Plugging it into OpenQA scripts is another issue, but an easy one. The hardware IOMMU support isn't relevant here, because we're talking about emulated devices, not physical ones.

**Comment by 0spinboson** [comment]
two other weird issues: 
I've now twice experienced the system becoming unresponsive to (USB-qube-attached) keyboard and mouse input, even though the qube didn't appear to have crashed. 

Also, when I pressed reset, the GPU took about 2 seconds to 'clean' up the old image from the OS that was no longer in memory or anything, so while it was POSTing. 

kernel 5.4.5, which works fine in my Qubes 4 installation, as does the GPU.

**Comment by marmarek** [comment]
> Is it normal for me to have to run fstrim -av manually? I just did so in dom0, got "trimmed 172GB", which seems a lot.

In most cases it is misleading message. Linux don't keep track of blocks discarded (or not allocated at all) across reboots.

**Comment by 0spinboson** [comment]
okay. I am running into another issue now though, that's very consistent. Namely, many (but not all) VM windows either not being drawn properly if the total area for a VM exceeds 3840x2160px. Most of the time this simple leads to part of the area remaining white/unfilled, much of the time the whole window won't refresh/update properly until I reduce the size again, and part of the time the VM window disappears entirely and crashes (looking at 'top' output inside the vm).

**Comment by JarrahG** [comment]
Current build issue: pungi-legacy does not exist in the fedora 31 repos. This only shows up at "make iso" when it is installing dependencies.

**Comment by fepitre** [comment]
Are you using this branch: https://github.com/QubesOS/qubes-installer-qubes-os/pull/34?

**Comment by mfp20** [comment]
@marmarek thank you. I've already seen this thread yesterday when I stomped on the lack of some salt componets during the R4.1 build process. I can't see a link to "4.1 from the October iso" referenced by @brendanhoar a few comments back... 

I've no urgency to have R4.1 installed. I need that in order to have FIO in dom0 and a few other things I have in mind, but it's not an urgent issue. I'm more concerned about having R4.1 to test stuff and report here for the project sake, not for myself. So, please, don't feel pushed...

**Comment by tricofait** [comment]
Hello,

Just to share my experience and to offer help testing if needed.  I've installed 4.1 20191227 iso ( just got an update BTW ). None of the backed up qubes restored displayed correctly ( based on templates cloned from Debian 10 and Fedora 30 and backed up on previous versions of qubes ). Decided to recreate them from scratch. 

Had problems displaying correctly libreoffice on newly created ( office sw machine ) cloned from Fedora 30, tried the two latest versions available for download, disabled hardware acceleration, safe mode, nothing worked. Then replaced the base template ( for office stuff ) with one cloned from Debian 10 and all goes well, and very fast displaying. Including GIMP that loads and performs faster.

I have recreated another qube with postgresql, dotnet core and visualstudio code for development and visually i do not have lag anymore with the scroll of the messages on the debug windows. 

I still have to migrate another qube ( testing environment for a website integration that runs on UBUNTU server 16.04 ) with apache mysql joomla to integrate with the devs that i make in dotnet. This qube will have some specific ports open to comunicate with the dev machine to simulate production conditions. 

I am happy with the change so far, and thinking of using 4.1 as of today, as the main OS ( i have another disk that boots with 4.0.2-rc3  that i am keeping ). I can do all that i used to do with the previous versions and faster. Please let me know if you need me to test anything particular.

All the best

**Comment by marmarek** [comment]
There is a new build here: https://openqa.qubes-os.org/tests/5493/asset/iso/Qubes-4.1-20200113-x86_64.iso
No big changes, besides some installer and salt fixes.

**Comment by GWeck** [comment]
See also my experiences with the 20191223 version in #5003

**Comment by GWeck** [comment]
I tried this new build, specifying manual partitioning, and ran again into the problems with partition sizes expressed as non-integer values in the kickstart file, causing the second phase of the installation, after reboot, to be aborted, just as described in #5003. The funny thing is that this time I specified partition sizes as GiB or MiB, and the installer always changes the given integer values to near-by non-integer values, e.g. 1 GiB oder 1024 MiB to 1023.98 MiB for the boot partition, and so on. These values are then in the kickstart file. I found no way to enter partition sizes in such a way that they stayed integer.

I truncated these value in the kickstart file to integer and then was able to start the system configuration manually.

**Comment by tricofait** [comment]
There was a dom0 update yesterday morning, that changed 13 packages.  
I do not know if it was related, but after that update all other templates were not able to update anymore. The message is connection refused on localhost. Tried to rollback the updates via dnf history and its id but it did not succeed. 

Since i had my most recent work there, tried to back it up just by using the visual files interface and copying it to the USB qube. It was somehow not possible. I ended up uploading it via SFTP, then booted with the old system on the other disk to continue working. ( so internet was working ).

Back to the old system for now because i have a deadline and not so much time to continue digging for solutions.

Thank you, just to keep you posted.

**Comment by 92VV3M42d3v8** [comment]
1.facing issue #5574 with 13 jan build.
2.Qubes Updater is not able to update debian and fedora in this build for me.
3.Why KeePassXC is version 2.4.3 here, why not 2.5.2
4. When we attach like sdb(usb) to a vm like vault, message appears that sdb1 is removed and we detach sdb, sdb1 reappears, Is that a expected behaviour?

**Comment by marmarek** [comment]
>  When we attach like sdb(usb) to a vm like vault, message appears that sdb1 is removed and we detach sdb, sdb1 reappears, Is that a expected behaviour?

Yes - this is because you can't attach both sdb (whole device) and sdb1 (individual partition) at the same time. Notifications could be improved though.

**Comment by marmarek** [comment]
> 3.Why KeePassXC is version 2.4.3 here, why not 2.5.2

It's installed what is in the distribution repository at the template build time. [Debian has 2.4.3](https://packages.debian.org/buster/keepassxc). Fedora only recently has 2.5.2 and template wasn't rebuilt since then.

>  2.Qubes Updater is not able to update debian and fedora in this build for me.

Any details about that? It works for me... Please create separate issue.

**Comment by asdfg45ui** [comment]
Sorry A little offtopic but @marmarek Isn't it 2.3.4 in Debian. And Default Vault VM is based on Fedora and still it's got 2.4.3

**Comment by marmarek** [comment]
Yes, that's what I've said in the last sentence. When you install updates, you should get it.

**Comment by JarrahG** [comment]
I've just built my own version of the 4.1 iso with the following patches:
* https://github.com/QubesOS/qubes-desktop-linux-kde-breeze/pull/3
* https://github.com/QubesOS/qubes-installer-qubes-os/pull/34
* https://github.com/QubesOS/qubes-anaconda/pull/4
* All Travis ones still open in this thread. 

In this ISO, it seems that a number of things fixed in the openQA versions are still a problem. The keyboard layout does not get passed to the VMs, and the grub.cfg file gets output in `/EFI/fedora/` not `/EFI/qubes/`. 

Are there PRs that I'm missing or is it something else?

**Comment by marmarek** [comment]
Are you sure you have up to date version of those branches? For example EFI dir wasn't there initially, but only pushed later (almost a month ago): https://github.com/QubesOS/qubes-anaconda/pull/4/commits/72856cce154e7115362b44ef5b4a6020fa19efb6
In openqa build I additionally have `stable-5.4` branch of `linux-kernel` repo, but it shouldn't matter for those problems.

**Comment by JarrahG** [comment]
> Are you sure you have up to date version of those branches? For
> example EFI dir wasn't there initially, but only pushed later (almost
> a month ago): QubesOS/qubes-anaconda@72856cc
> <https://github.com/QubesOS/qubes-anaconda/commit/72856cce154e7115362b44ef5b4a6020fa19efb6>
>
Missed this one. I used the pull request above on top of
QubesOS/qubes-anaconda rather than fepitre/qubes-anaconda. That would
explain the difference.
>

**Comment by 0spinboson** [comment]
> Reported upstream: https://bugzilla.redhat.com/show_bug.cgi?id=1785489

geez, they're certainly not in a rush there.

**Comment by marmarek** [comment]
> > Reported upstream: https://bugzilla.redhat.com/show_bug.cgi?id=1785489
> 
> geez, they're certainly taking their time.

I've just got this notification:
> This package has changed maintainer in the Fedora.
Reassigning to the new maintainer of this component.

**Comment by 92VV3M42d3v8** [comment]
31st Jan build crashes while install with Anaconda errors.

**Comment by firstabor** [comment]
Trying to install the qubes-4.1-20200113
Qubes boot option isnt recognized i tried to trouble shoot like described in the UEFI troubleshooting Doc but i dont know how to write the boot entry.

When i "rescue a qube" from usb and press 1 to continue i get this error

![1581030216561136045852778237779](https://user-images.githubusercontent.com/57239935/73986216-b6ffa600-4945-11ea-926e-5b3720bb5882.jpg)

**Comment by 0rb677** [comment]
can't boot 4.1 ISO via kexec. horrible screen artefacts on the screen :)
its stuck and won't boot
![DSC_0079](https://user-images.githubusercontent.com/60626673/74088254-e9de9280-4a48-11ea-91ed-2da041be9c66.JPG)
![DSC_0080](https://user-images.githubusercontent.com/60626673/74088255-eba85600-4a48-11ea-9667-0b7b146f2a35.JPG)

> There is a new build here: https://openqa.qubes-os.org/tests/5493/asset/iso/Qubes-4.1-20200113-x86_64.iso
> No big changes, besides some installer and salt fixes.

**Comment by 0rb677** [comment]
@marmarek is there any new 4.1 iso for testing?

**Comment by arno01** [comment]
> i finally managed to go to the "Rescue a Qubes" and now after pressing 1 (continue)
> i get
> You don't have any Linux partitions. Rebooting.

This is a known issue https://github.com/QubesOS/qubes-issues/issues/5609
A really quick ad-hoc workaround would be to rename all ``find_existing_installations`` to ``_find_existing_installations`` in the ``/usr/lib*/python*/site-packages/pyanaconda/rescue.py`` file and then re-run anaconda rescue: ``anaconda --rescue --lang us``

**Comment by JarrahG** [comment]
Would it be possible to get a copy of the current build config used on OpenQA @marmarek? I'd like to test some of the currently unmerged features (GUI VM and AMD suspend) but can't get a build to compile.

**Comment by tricofait** [comment]
Hello  all,

Back to v4.1 as main installation, ( fresh install ) now with an nvme SSD ( all is so so quick... above all video ) really amazing job!
All the problems i had at the beginning of the year are gone with this most recent version iso.

I think i have all the updates, not really sure about all firmware, because with journalctl -axb ( boot of dom0 ) i get these annoying messages in bold below that i would like to put away if possible being this my main installation now ( these messages do not show on 4.0.3 )

<pre>Mar 03 04:05:25 dom0 kernel: Spectre V1 : Mitigation: usercopy/swapgs barriers and __user pointer sanitization
Mar 03 04:05:25 dom0 kernel: Spectre V2 : Mitigation: Full generic retpoline
Mar 03 04:05:25 dom0 kernel: Spectre V2 : Spectre v2 / SpectreRSB mitigation: Filling RSB on context switch
Mar 03 04:05:25 dom0 kernel: Spectre V2 : Enabling Restricted Speculation for firmware calls
Mar 03 04:05:25 dom0 kernel: Spectre V2 : mitigation: Enabling conditional Indirect Branch Prediction Barrier
Mar 03 04:05:25 dom0 kernel: Spectre V2 : User space: Mitigation: STIBP via seccomp and prctl
Mar 03 04:05:25 dom0 kernel: Speculative Store Bypass: Mitigation: Speculative Store Bypass disabled via prctl and seccomp
Mar 03 04:05:25 dom0 kernel: MDS: Mitigation: Clear CPU buffers
Mar 03 04:05:25 dom0 kernel: Freeing SMP alternatives memory: 28K
Mar 03 04:05:25 dom0 kernel: <font color="#D7D75F"><b>cpu 0 spinlock event irq 121</b></font>
Mar 03 04:05:25 dom0 kernel: VPMU disabled by hypervisor.
Mar 03 04:05:25 dom0 kernel: Performance Events: unsupported p6 CPU model 78 no PMU driver, software events only.
Mar 03 04:05:25 dom0 kernel: rcu: Hierarchical SRCU implementation.
Mar 03 04:05:25 dom0 kernel: NMI watchdog: Perf NMI watchdog permanently disabled
Mar 03 04:05:25 dom0 kernel: smp: Bringing up secondary CPUs ...
Mar 03 04:05:25 dom0 kernel: installing Xen timer for CPU 1
Mar 03 04:05:25 dom0 kernel: <font color="#D7D75F"><b>cpu 1 spinlock event irq 133</b></font>
Mar 03 04:05:25 dom0 kernel: <font color="#D7D75F"><b>MDS CPU bug present and SMT on, data leak possible. See https://www.kernel.org/doc/html/latest/admin-guide/hw-vuln/mds.html for more details.</b></font>
</pre>

Can i do something to fix this?

Thank you so much in advance, great job on 4.1!

**Comment by JarrahG** [comment]
> Mar 03 04:05:25 dom0 kernel: <font color="#D7D75F"><b>MDS CPU bug present and SMT on, data leak possible. See https://www.kernel.org/doc/html/latest/admin-guide/hw-vuln/mds.html for more details.</b></font>
> </pre>
>
> Can i do something to fix this?

This is part of the Intel CPU mitigations which have come out over the
last year and a bit. Adding `nosmt` to the kernel command line in
`/etc/default/grub` should resolve the warning. If this doesn't work,
disable SMT/hyperthreading in BIOS.

I believe this was added as a default in 4.0 at some point, in which
case this may be a regression. However, I may also have just added it
myself when it was recommended.

**Comment by tricofait** [comment]
Thank you very much @JarrahG ,

I added the nosmt boot parameter, i have a different and verbose warning: 
<pre>Mar 04 08:19:47 dom0 kernel: Spectre V1 : Mitigation: usercopy/swapgs barriers and __user pointer sanitization
Mar 04 08:19:47 dom0 kernel: Spectre V2 : Mitigation: Full generic retpoline
Mar 04 08:19:47 dom0 kernel: Spectre V2 : Spectre v2 / SpectreRSB mitigation: Filling RSB on context switch
Mar 04 08:19:47 dom0 kernel: Spectre V2 : Enabling Restricted Speculation for firmware calls
Mar 04 08:19:47 dom0 kernel: Spectre V2 : mitigation: Enabling conditional Indirect Branch Prediction Barrier
Mar 04 08:19:47 dom0 kernel: Spectre V2 : User space: Mitigation: STIBP via seccomp and prctl
Mar 04 08:19:47 dom0 kernel: Speculative Store Bypass: Mitigation: Speculative Store Bypass disabled via prctl and seccomp
Mar 04 08:19:47 dom0 kernel: MDS: Mitigation: Clear CPU buffers
Mar 04 08:19:47 dom0 kernel: Freeing SMP alternatives memory: 28K
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>cpu 0 spinlock event irq 121</b></font>
Mar 04 08:19:47 dom0 kernel: VPMU disabled by hypervisor.
Mar 04 08:19:47 dom0 kernel: Performance Events: unsupported p6 CPU model 78 no PMU driver, software events only.
Mar 04 08:19:47 dom0 kernel: rcu: Hierarchical SRCU implementation.
Mar 04 08:19:47 dom0 kernel: NMI watchdog: Perf NMI watchdog permanently disabled
Mar 04 08:19:47 dom0 kernel: smp: Bringing up secondary CPUs ...
Mar 04 08:19:47 dom0 kernel: installing Xen timer for CPU 1
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>------------[ cut here ]------------</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>WARNING: CPU: 0 PID: 1 at drivers/xen/events/events_base.c:1111 unbind_from_irqhandler+0x36/0x40</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>Modules linked in:</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>CPU: 0 PID: 1 Comm: swapper/0 Not tainted 5.4.5-1.qubes.x86_64 #1</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>Hardware name: Purism Librem 13 v2/Librem 13 v2, BIOS 4.9-Purism-2 11/13/2019</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>RIP: e030:unbind_from_irqhandler+0x36/0x40</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>Code: fd e8 2e 04 b3 ff 48 85 c0 74 1f 48 8b 40 10 48 83 78 08 00 74 14 89 ef 4c 89 e6 e8 c4 d6 b2 ff 89 ef 5d 41 5c e9 ba fe ff ff &lt;0f&gt; 0b 5d 41 5c c3 0f 1f 40</b></font><span style="background-color:#FFFFFF"><font color="#000000">&gt;</font></span>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>RSP: e02b:ffffc90000e17d10 EFLAGS: 00010246</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>RAX: 0000000000000000 RBX: 0000000000000001 RCX: 0000000000000fff</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>RDX: 0000000000000028 RSI: 00000000ffffffff RDI: ffffffff8265b680</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>RBP: 00000000ffffffff R08: 0000000000001000 R09: 0000000000000238</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>R10: 0000000000000000 R11: ffffffff8265b688 R12: 0000000000000000</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>R13: 0000000000000001 R14: 0000000000000000 R15: ffffffff810e5180</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>FS:  0000000000000000(0000) GS:ffff888185200000(0000) knlGS:0000000000000000</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>CS:  e030 DS: 0000 ES: 0000 CR0: 0000000080050033</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>CR2: ffffc90000c2a000 CR3: 000000000260a000 CR4: 0000000000040660</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>Call Trace:</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> xen_uninit_lock_cpu+0x28/0x62</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> xen_pv_cpu_die+0x8e/0xb0</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> takedown_cpu+0x9c/0xe0</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> cpuhp_invoke_callback+0x8d/0x550</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> _cpu_up+0x11a/0x130</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> do_cpu_up+0x7e/0xc0</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> smp_init+0xc1/0x189</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> kernel_init_freeable+0x169/0x34a</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> ? rest_init+0xaa/0xaa</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> kernel_init+0xa/0x106</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b> ret_from_fork+0x35/0x40</b></font>
Mar 04 08:19:47 dom0 kernel: <font color="#D7D75F"><b>---[ end trace 1dd29ab08403a57d ]---</b></font>
</pre>

What can i do to help on this one?
I am fully available to solve this, just let me know.

Thank you very much.

**Comment by JarrahG** [comment]
> Mar 04 08:19:47 dom0 kernel: smp: Bringing up secondary CPUs ...
> Mar 04 08:19:47 dom0 kernel: installing Xen timer for CPU 1
>
> What can i do to help on this one?
> I am fully available to solve this, just let me know.
>
> Thank you very much.
>
Complete guess here, but this may be because we just turned off SMT in
Linux (dom0) but not Xen/BIOS.

Try either adding `smt=off` to the Xen commandline or disabling it in BIOS.

**Comment by marmarek** [comment]
Don't you have `smt=off` on Xen cmdline by default? Check `xl info`.
Also note that the method Xen use to disable SMT doesn't play nicely with Linux trying to do the same (or even detect SMT presence). See https://github.com/speed47/spectre-meltdown-checker/issues/343 for some more details.

**Comment by marmarek** [comment]
> Would it be possible to get a copy of the current build config used on OpenQA @marmarek?

https://gist.github.com/marmarek/f0e225b593650212108f1ea6974f5914

**Comment by tricofait** [comment]
> Don't you have smt=off on Xen cmdline by default?
> @marmarek
Yes i do. The comment above on March 3rd had this parameter already on Xen command line and the warning below came out:
**MDS CPU bug present and SMT on, data leak possible. See https://www.kernel.org/doc/html/latest/admin-guide/hw-vuln/mds.html for more details.**

The second comment here on March the 4th, was after the suggestion of @JarrahG to additionally insert nosmt on the Linux ( dom0 ) command line. It is still there. I removed it for testing and i got back to the situation on the first comment of March 3rd. Then i put it back again.

I did a little research, found the [events_base.c] (https://elixir.bootlin.com/linux/v5.4.25/source/drivers/xen/events/events_base.c#L1111) and the kernel on previous versions did not change on that particular line **i am very new to this**.
 
Following the source, what seems to me is that when the kernel is trying **[cpu 1 spinlock event irq 133]** (https://elixir.bootlin.com/linux/v5.4.25/source/drivers/xen/events/events_base.c#L1109), when installing Xen timer for CPU 1, it gets no info, and so, therefore the WARN ( from today's boot ):
<pre>Mar 14 05:00:20 dom0 kernel: installing Xen timer for CPU 1
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>------------[ cut here ]------------</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>WARNING: CPU: 0 PID: 1 at drivers/xen/events/events_base.c:1111 unbind_from_irqhandler+0x36/0x40</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>Modules linked in:</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>CPU: 0 PID: 1 Comm: swapper/0 Not tainted 5.4.5-1.qubes.x86_64 #1</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>Hardware name: Purism Librem 13 v2/Librem 13 v2, BIOS 4.9-Purism-2 11/13/2019</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>RIP: e030:unbind_from_irqhandler+0x36/0x40</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>Code: fd e8 2e 04 b3 ff 48 85 c0 74 1f 48 8b 40 10 48 83 78 08 00 74 14 89 ef 4c 89 e6 e8 c4 d6 b2 ff 89 ef 5d 41 5c e9 ba fe ff ff &lt;0f&gt; 0b 5d 41 5c c3 0f 1f 40</b></font><span style="background-color:#FFFFFF"><font color="#000000">&gt;</font></span>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>RSP: e02b:ffffc90000e17d10 EFLAGS: 00010246</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>RAX: 0000000000000000 RBX: 0000000000000001 RCX: 0000000000000fff</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>RDX: 0000000000000028 RSI: 00000000ffffffff RDI: ffffffff8265b680</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>RBP: 00000000ffffffff R08: 0000000000001000 R09: 0000000000000238</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>R10: 0000000000000000 R11: ffffffff8265b688 R12: 0000000000000000</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>R13: 0000000000000001 R14: 0000000000000000 R15: ffffffff810e5180</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>FS:  0000000000000000(0000) GS:ffff888185200000(0000) knlGS:0000000000000000</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>CS:  e030 DS: 0000 ES: 0000 CR0: 0000000080050033</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>CR2: ffffc90000c2a000 CR3: 000000000260a000 CR4: 0000000000040660</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>Call Trace:</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> xen_uninit_lock_cpu+0x28/0x62</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> xen_pv_cpu_die+0x8e/0xb0</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> takedown_cpu+0x9c/0xe0</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> cpuhp_invoke_callback+0x8d/0x550</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> _cpu_up+0x11a/0x130</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> do_cpu_up+0x7e/0xc0</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> smp_init+0xc1/0x189</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> kernel_init_freeable+0x169/0x34a</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> ? rest_init+0xaa/0xaa</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> kernel_init+0xa/0x106</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b> ret_from_fork+0x35/0x40</b></font>
Mar 14 05:00:20 dom0 kernel: <font color="#D7D75F"><b>---[ end trace 1398e263b26a61a8 ]---</b></font>
</pre>
Is shown before the IRQ ( 113 ) i think, can be freed and unbound.

How can i setup an environment to debug this and possibly trace line by line ( with screenshots ) what is happening to be sure and possibly bring better info?

I have been using the version ** https://openqa.qubes-os.org/tests/5493/asset/iso/Qubes-4.1-20200113-x86_64.iso** with all the updates and all seems to be working nicely besides the suspend to RAM. 

It suspends well, but upon awaking, some VMs based on fedora 30 get confused because they did not suspend properly and they are still trying to do it ( upon awake ) by the message. The error condition persisted even after shutting all the f30 vm's. They would not boot. The workaround that seemed to work to clean the status was shutting all the vms down ( not dom0, of course ), then boot up with sys-usb ( ** _that was not being used when the the suspend was done_ ** ). The boot of sys-usb succeeded ( somehow it cleaned some sort of state ( i suspect related to the f30 template ) on dom0 ). After that successful boot of sys-usb, all f30 vms ( sys-net, sys-firewall ) were able to boot up and worked normally.

Thank you so much, great project! Count on me, i am here to help.

**Comment by zraffer** [comment]
Have been using F31 in R4.0 for a month without troubles. Are there any obstacles not to release it now?

Also see rather old F30 VM in the current R4.1 ISO build being tested at the OpenQA server. Are there any reasons to test coming version of the OS against the old F30 VM whereas we have new F31 VM already?

**Comment by ghty968** [comment]
Why not to base dom0 on fedora-32 in Qubes 4.1

**Comment by osresearch** [comment]
> "no-real-mode" is/was mostly relevant for people booting through kexec (linuxboot, heads,...)
> if this is no longer required (because xen figures out what mode the cpu is in on its own), good.
> if it is still required (but gone), bad.

`no-real-mode` is still there, but is in a set of parameters that are only used by the very early argument parsing code.  I've submitted a patch to xen-devel to silence the warning about those parameters. https://lists.xenproject.org/archives/html/xen-devel/2020-08/msg00521.html

**Comment by marmarek** [comment]
Superseded by https://github.com/QubesOS/qubes-issues/issues/5763
